content/python/concepts/command-line-arguments/command-line-arguments.md

```diff
@@ -0,0 +1,101 @@
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
+Tags:
+  - 'Input'
+  - 'Arguments'
+  - 'Bash/Shell'
+  - 'Command Line'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+Python offers several methods of parsing command line arguments that are used when launching a program. The user has the option of passing various parameters to the program by adding them after the program name. These parameters can be accessed through various modules. The parameter names are left up to the programmer. By convention the parameter `-h` or `--help` (and `/h` and `/help` in Windows) is reserved to print a help message to the console listing the various command line options recognized by the program.
+
+Arguments are passed differently depending on the operating system:
+
+- Unix-like arguments are a single letter preceded by a dash (`-h`) or a word preceded by two dashes (`--help`).
+- Windows arguments are a letter or a whole word preceded by a slash (`/h` or `/help`).
+
+## Parsing Command Line Arguments With `sys.argv`
+
+The `sys` module offers the most straightforward method of accessing command line arguments, but it also requires the programmer to do most of the work interpreting them. With the `sys` module, the arguments are passed along in a simple list structure named `sys.argv`.
+
+The first item in the list, `sys.argv[0]` is the name used to launch the Python program, along with the path used. Each subsequent item is a space-delimited argument from the command line used to launch the program. If an argument requires embedded spaces, it needs to be enclosed in quotes to be parsed correctly.
+
+### Example
+
+This prints the script name followed by a list of the command line arguments passed on to the program.
+
+```python
+import sys
+
+print(f"The program name used to launch me is {sys.argv[0]}.")
+print("I was passed the following arguments:")
+for arg in sys.argv[1:]:
+  print(arg)
+```
+
+If this is named **test.py** it can be launched with the following result:
+
+```bash
+$ test.py --arg1 --arg2 "arg 3"
+The program name used to launch me is test.py.
+I was passed the following arguments:
+--arg1
+--arg2
+arg 3
+```
+
+## Parsing Command Line Arguments With `getopt()`
+
+Using `getopt()` requires importing both the `sys` and `getopt` modules to work. With `getopt`, parameter validation is added to command line parsing. This is done by passing a list of the command line arguments themselves, a string of short (one character) options, and a list of long (full word) options. To indicate that the option requires a value to be passed along with it, the short option is followed by a colon (`:`), and the long option is followed by an equals sign (`=`). So to set up options for "help", "argument" and a "value" that requires an additional passed value, the short options would be a string like this, `"hav:"` and the long options would be a list like this, `["help","argument","value="]`.
```

Comment:
```suggestion
Using `getopt()` requires importing both the `sys` and `getopt` modules to work. With `getopt`, parameter validation is added to command line parsing. This is done by passing a list of the command line arguments themselves, a string of short (one character) options, and a list of long (full word) options. To indicate that the option requires a value to be passed along with it, the short option is followed by a colon (`:`), and the long option is followed by an equals sign (`=`).

To set up options for "help", "argument", and a "value" that requires an additional passed value, would look like this:

- The short options would be a string like `"hav:"`.
- The long options would be a list like `["help","argument","value="]`.
```

Reply:
@SSwiniarski thinking this could be broken down like so

Reply:
Makes sense.